tensorflow/core/kernels/segment_reduction_ops.cc

```diff
@@ -215,13 +215,13 @@ REGISTER_COMPLEX_CPU_KERNELS_ALL(complex128);
 #undef REGISTER_REAL_CPU_KERNELS_ALL
 #undef REGISTER_COMPLEX_CPU_KERNELS_ALL
 
-// Similar to SegmentReductionOp but can handle unsorted segment definitions and
-// specifying size of output.
+// base class for UnsortedSegmentMax and UnsortedSegmentSum
 template <typename Device, class T, class Index>
-class UnsortedSegmentSumOp : public OpKernel {
- public:
-  explicit UnsortedSegmentSumOp(OpKernelConstruction* context)
-      : OpKernel(context) {}
+class UnsortedSegmentOpBase : public OpKernel {
+  public:
+    explicit UnsortedSegmentOpBase(OpKernelConstruction* context,
+                                   bool is_max)
```

Comment:
Instead of passing an is_max argument, why not create a virtual function to initialize the output, similar to the reduction function ?